frame/executive/src/lib.rs

```diff
@@ -286,13 +283,8 @@ where
 		assert!(
 			n > System::BlockNumber::zero()
 			&& <frame_system::Module<System>>::block_hash(n - System::BlockNumber::one()) == *header.parent_hash(),
-			"Parent hash should be valid."
+			"Parent hash should be valid.",
 		);
-
-		// Check that transaction trie root represents the transactions.
-		let xts_root = extrinsics_root::<System::Hashing, _>(&block.extrinsics());
-		header.extrinsics_root().check_equal(&xts_root);
-		assert!(header.extrinsics_root() == &xts_root, "Transaction trie root must be valid.");
```

Comment:
just to be sure, there wasn't need for this to be moved in `final_checks` except for avoiding duplicated code, isn't it ?

Reply:
Removes duplicate code and work. We calculate the extrinsics root while finalizing the block. This means we don't need to calculate here again.

Reply:
yes all clear